tests/sjs-junit/test/org/scalajs/testsuite/jsinterop/NonNativeJSTypeTestScala3.scala

```diff
@@ -0,0 +1,56 @@
+package org.scalajs.testsuite.jsinterop
+
+import org.junit.Assert.*
+import org.junit.Test
+
+import scala.scalajs.js
+import scala.scalajs.js.annotation.*
+
+class NonNativeJSTypeTestScala3 {
+  import NonNativeJSTypeTestScala3.*
+
+  @Test
+  def overloadWithVarargOfGenericType(): Unit = {
+    class OverloadWithVarargOfGenericType extends js.Object {
+      def overloaded(x: Int): Int = x
+      def overloaded(xs: (Int, Int)*): Int = xs.size
+    }
+
+    val obj = new OverloadWithVarargOfGenericType
+    assertEquals(5, obj.overloaded(5))
+    assertEquals(1, obj.overloaded((5, 6)))
+    assertEquals(2, obj.overloaded((1, 2), (3, 4)))
+  }
+
+  @Test
+  def overloadWithVarargOfValueClass(): Unit = {
+    class OverloadWithVarargOfValueClass extends js.Object {
+      def overloaded(x: Int): Int = x
+      def overloaded(xs: VC*): Int = xs.size
+    }
+
+    val obj = new OverloadWithVarargOfValueClass
+    assertEquals(5, obj.overloaded(5))
+    assertEquals(1, obj.overloaded(new VC(5)))
+    assertEquals(2, obj.overloaded(new VC(5), new VC(6)))
+  }
+
+  @Test
+  def overloadWithVarargOfGenericValueClass(): Unit = {
```

Comment:
We should also test opaque-type aliases.

Reply:
Added.